app/components/dex-indicator.jsx

```diff
@@ -0,0 +1,20 @@
+import { connect } from 'react-redux';
+
+export function DexIndicator ({ dex, profile }) {
+
+  if ((dex && !dex.shiny) || profile) {
+    return null;
+  }
+
+  return (
+    <div className="dex-indicator" title="shiny">
+      <i className="fa fa-star" />
+    </div>
+  );
+}
+
+function mapStateToProps ({ currentUser }) {
+  return { username: currentUser };
```

Comment:
you don't use `username` in the component so its unnecessary. which makes `mapStateToProps` unnecessary. which makes `connect` unnecessary.